immich-app/immich
# feat(server,web): server config

Fixes #3994

In this PR:
- Add map settings to admin UI
- Add option to change the tile map server url
- Expose non-boolean config via a new endpoint `GET /server-info/config`, which can be used with `GET /server-info/features`. Maybe merge the two endpoints in the future?
- Expose `oauth.buttonText`, so it can be used (again) on the login page
- Move the login page message to server config. In the future this can be saved/set in admin settings as well.
- Sort admin settings alphabetically

Tested scenarios:
- Disable map, map tab disappears in sidebar (after a refresh), map is hidden in the detail panel
- Set custom OAuth button text, view it in on the login page

Future plans:
- Use websockets to emit changes in feature flags / configurations
- Add new admin setting option for login page message (instead of env variable)
- Add more e2e tests to verify system config / server config is working correctly.
